lib/ReactViews/Map/Panels/SharePanel/ShareUrl/ShareUrl.tsx

```diff
@@ -0,0 +1,153 @@
+import React, {
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useState,
+  PropsWithChildren
+} from "react";
+import { useTranslation } from "react-i18next";
+import styled from "styled-components";
+
+import Terria from "../../../../../Models/Terria";
+import ViewState from "../../../../../ReactViewModels/ViewState";
+
+import Spacing from "../../../../../Styled/Spacing";
+import { TextSpan } from "../../../../../Styled/Text";
+
+import { buildShareLink, buildShortShareLink } from "../BuildShareLink";
+import { ShareUrlWarning } from "./ShareUrlWarning";
+import Clipboard from "../../../../Clipboard";
+import Input from "../../../../../Styled/Input";
+import {
+  Category,
+  ShareAction
+} from "../../../../../Core/AnalyticEvents/analyticEvents";
+
+interface IShareUrlProps {
+  terria: Terria;
+  viewState: ViewState;
+  includeStories: boolean;
+  shouldShorten: boolean;
+  theme: "light" | "dark";
+  inputTheme?: "light" | "dark";
+  rounded?: boolean;
+  callback?: () => void;
+}
+
+export interface IShareUrlRef {
+  url: string;
+  shorteningInProgress: boolean;
+}
+
+export const ShareUrl = forwardRef<
+  IShareUrlRef,
+  PropsWithChildren<IShareUrlProps>
+>(
+  (
+    {
+      terria,
+      viewState,
+      includeStories,
+      shouldShorten,
+      children,
+      theme,
+      inputTheme,
+      rounded,
+      callback
+    },
+    forwardRef
+  ) => {
+    const { t } = useTranslation();
+
+    const [shareUrl, setShareUrl] = useState("");
+    const [shorteningInProgress, setShorteningInProgress] = useState(false);
+    const [placeholder, setPlaceholder] = useState<string>();
+
+    useImperativeHandle(
+      forwardRef,
+      () => ({
+        url: shareUrl,
+        shorteningInProgress: shorteningInProgress
+      }),
+      [forwardRef, shareUrl, shorteningInProgress]
+    );
+
+    const buildUnshortenderUrl = (
+      terria: Terria,
+      viewState: ViewState,
+      includeStories: boolean
+    ) => {
+      return buildShareLink(terria, viewState, {
+        includeStories
+      });
+    };
```

Comment:
I think this function should be removed, and `buildShareLink` should be called directly

Reply:
Good catch, implemented.